src/sumBy.test-d.ts

```diff
@@ -0,0 +1,85 @@
+import { sumBy } from "./sumBy";
+import { expectTypeOf } from "vitest";
+import { pipe } from "./pipe";
+
+test("empty array", () => {
+  const result1 = sumBy([], BigInt);
+  expectTypeOf(result1).toEqualTypeOf<0>();
+
+  const result2 = sumBy([], toNumber);
+  expectTypeOf(result2).toEqualTypeOf<number>();
+});
+
+describe("numbers", () => {
+  test("arbitrary arrays", () => {
+    const result = sumBy([] as Array<unknown>, toNumber);
+    expectTypeOf(result).toEqualTypeOf<number>();
+  });
+
+  test("arbitrary readonly arrays", () => {
+    const result = sumBy([] as ReadonlyArray<unknown>, toNumber);
+    expectTypeOf(result).toEqualTypeOf<number>();
+  });
+
+  test("arbitrary non-empty arrays", () => {
+    const result = sumBy([1, 2] as [unknown, ...Array<unknown>], toNumber);
+    expectTypeOf(result).toEqualTypeOf<number>();
+  });
+
+  test("consts", () => {
+    const result = sumBy([1, 2, 3] as const, toNumber);
+    expectTypeOf(result).toEqualTypeOf<number>();
+  });
+
+  test("fixed-size tuples", () => {
+    const result = sumBy([1, 2] as [unknown, unknown], toNumber);
+    expectTypeOf(result).toEqualTypeOf<number>();
+  });
+});
+
+describe("bigints", () => {
+  test("arbitrary arrays", () => {
+    const result = sumBy([] as Array<unknown>, toBigInt);
+    expectTypeOf(result).toEqualTypeOf<bigint | 0>();
+  });
+
+  test("arbitrary readonly arrays", () => {
+    const result = sumBy([] as ReadonlyArray<unknown>, toBigInt);
+    expectTypeOf(result).toEqualTypeOf<bigint | 0>();
+  });
+
+  test("arbitrary non-empty arrays", () => {
+    const result = sumBy([1n, 2n] as [unknown, ...Array<unknown>], toBigInt);
+    expectTypeOf(result).toEqualTypeOf<bigint>();
+  });
+
+  test("consts", () => {
+    const result = sumBy([1n, 2n, 3n] as const, toBigInt);
+    expectTypeOf(result).toEqualTypeOf<bigint>();
+  });
+
+  test("fixed-size tuples", () => {
+    const result = sumBy([1n, 2n] as [unknown, unknown], toBigInt);
+    expectTypeOf(result).toEqualTypeOf<bigint>();
+  });
+});
+
+describe("dataLast", () => {
```

Comment:
Add tests for the empty case for data-last to test that the types are inferred correctly "through" the pipe.

Reply:
Two empty cases were added for stricter